specification/connector/_types/SyncJob.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Id } from '@_types/common'
+import { long } from '@_types/Numeric'
+import { DateTime } from '@_types/Time'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import {
+  ConnectorConfiguration,
+  FilteringConfig,
+  IngestPipelineParams,
+  SyncStatus
+} from './Connector'
+
+interface SyncJobConnectorReference {
+  configuration: ConnectorConfiguration
+  filtering: FilteringConfig
+  id: Id
+  index_name: string
+  language?: string
+  pipeline?: IngestPipelineParams
+  service_type: string
```

Comment:
`service type` seems to be [optional](https://github.com/elastic/elasticsearch/blob/eb6af0e6b549d0810d8b89791556bc791b766652/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java#L103) as well. other than this, LGTM :D

Reply:
For connector yes it can be optional. But you do need to have the `service_type` set to a non-null value before starting a sync job for a given connector. So we can expect this value to always be present.

I do see that our [sync job parser for its connector property](https://github.com/elastic/elasticsearch/blob/91d3bb026fa8272bb847e84f85cc18f6f93ebc2a/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java#L377) in ES is lacking with that respect  I'm creating an issue to fix that https://github.com/elastic/enterprise-search-team/issues/7344